csharp/src/Ice/Proxy.cs

```diff
@@ -246,13 +263,29 @@ public static class Proxy
         /// <param name="progress">Sent progress provider.</param>
         /// <param name="cancel">A cancellation token that receives the cancellation requests.</param>
         /// <returns>A task holding the response frame.</returns>
-        public static Task<IncomingResponseFrame> InvokeAsync(
+        public static async Task<IncomingResponseFrame> InvokeAsync(
             this IObjectPrx proxy,
             OutgoingRequestFrame request,
             bool oneway = false,
             IProgress<bool>? progress = null,
-            CancellationToken cancel = default) =>
-            InvokeWithInterceptorsAsync(proxy, request, oneway, synchronous: false, progress, cancel);
+            CancellationToken cancel = default)
+        {
+            CancellationTokenSource? cancellationSource = cancel.CanBeCanceled ?
```

Comment:
I don't understand this code.
We already combined the caller's cancellation token into the request's cancellation token. Why create another linked source token?
And it's probably better to keep this method non-async.

Reply:
I got rid of the CancelationToken parameter in Invoke and InvokeAsync, the user cancelation token if any must be supplied when the outgoing request frame is created